README.md
# Awesome - Slider

A highly customisable Rounded Rectangular thumb shaped slider for flutter projects which has shadow and child property.

![Example 01](screenfetch/squareThumb.gif) ![Example 02](screenfetch/rRThumb.gif)
![Example 03](screenfetch/circle.gif)

## Getting Started

- [Installation](#installation)
- [Basic Usage](#basic-usage)
- [Example Demo](#example-demo)
- [Awesome Slider parameters](#awesome-slider-parameters)


### Installation

Add

```bash
awesome_slider : ^0.1.0
```
to your pubspec.yaml, and run


```bash
flutter packages get
```
in your project's root directory.

### Basic Usage

Import it to your project file

```dart
import 'package:awesome_slider/awesome_slider.dart';
```
and use Widget

```dart
AwesomeSlider()
```


### Example Demo


![demo image for square](screenfetch/squareThumb.gif)
<details>
  <summary>Code example (click to expand)</summary>

```dart
import 'package:flutter/material.dart';
import 'package:awesome_slider/awesome_slider.dart';

void main() => runApp(MainApp());

class MainApp extends StatefulWidget {
  @override
  _MainAppState createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  double valueOnTextWidget = 0.0;
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Container(
          decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: [Colors.lightGreen, Colors.green],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              stops: [0.0, 0.6],
            ),
          ),
          child: Column(
            children: <Widget>[
              SizedBox(
                height: 250.0,
              ),
              Text(
                valueOnTextWidget.round().toString(),
                style: TextStyle(color: Colors.white, fontSize: 150.0),
              ),
              SizedBox(
                height: 80.0,
              ),
              AwesomeSlider(
                value: valueOnTextWidget,
                min: 0.0,
                max: 100.0,
                thumbColor: Colors.lightGreen,
                roundedRectangleThumbRadius: 0.0,
                topLeftShadow: true,
                topLeftShadowColor: Colors.lightGreen.shade300,
                topLeftShadowBlur: MaskFilter.blur(BlurStyle.normal, 8.0),
                bottomRightShadow: true,
                bottomRightShadowColor: Colors.black.withOpacity(0.5),
                bottomRightShadowBlur: MaskFilter.blur(BlurStyle.normal, 7.0),
                activeLineStroke: 2.0,
                activeLineColor: Colors.greenAccent,
                inactiveLineColor: Colors.white,
                child: Center(
                    child: Text(
                  'Slide Me',
                  style: TextStyle(
                    color: Colors.white.withOpacity(0.7),
                    fontSize: 20.0,
                    fontWeight: FontWeight.bold,
                  ),
                )),
                onChanged: (double value) {
                  setState(() {
                    valueOnTextWidget = value;
                  });
                },
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```
</details>


![demo image for Rounded square](screenfetch/rRThumb.gif)
<details>
  <summary>Code example (click to expand)</summary>

```dart
import 'package:flutter/material.dart';
import 'package:awesome_slider/awesome_slider.dart';

void main() => runApp(MainApp());

class MainApp extends StatefulWidget {
  @override
  _MainAppState createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  double valueOnTextWidget = 0.0;
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Container(
          decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: [
                Colors.lightBlueAccent,
                Color(0xFF100887).withOpacity(0.95)
              ],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              stops: [0.0, 0.5],
            ),
          ),
          child: Column(
            children: <Widget>[
              SizedBox(
                height: 250.0,
              ),
              Text(
                valueOnTextWidget.round().toString(),
                style: TextStyle(color: Colors.white, fontSize: 150.0),
              ),
              SizedBox(
                height: 80.0,
              ),
              AwesomeSlider(
                value: valueOnTextWidget,
                min: 0.0,
                max: 100.0,
                thumbColor: Color(0xFF100887),
                roundedRectangleThumbRadius: 25.0,
                thumbSize: 100.0,
                topLeftShadow: true,
                topLeftShadowColor: Colors.lightBlueAccent,
                topLeftShadowBlur: MaskFilter.blur(BlurStyle.normal, 11.0),
                bottomRightShadow: true,
                bottomRightShadowColor: Colors.white.withOpacity(0.5),
                bottomRightShadowBlur: MaskFilter.blur(BlurStyle.normal, 11.0),
                activeLineStroke: 2.0,
                activeLineColor: Colors.blueAccent,
                inactiveLineColor: Colors.white,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    Icon(
                      Icons.arrow_back_ios,
                      color: Colors.white,
                      size: 28.0,
                    ),
                    SizedBox(width: 10.0),
                    Icon(
                      Icons.arrow_forward_ios,
                      color: Colors.white,
                      size: 28.0,
                    )
                  ],
                ),
                onChanged: (double value) {
                  setState(() {
                    valueOnTextWidget = value;
                  });
                },
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```
</details>



![demo image for circular](screenfetch/circle.gif)
<details>
  <summary>Code example (click to expand)</summary>

```dart
import 'package:flutter/material.dart';
import 'package:awesome_slider/awesome_slider.dart';

void main() => runApp(MainApp());

class MainApp extends StatefulWidget {
  @override
  _MainAppState createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  double valueOnTextWidget = 0.0;
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Container(
          decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: [Colors.redAccent.shade200, Color(0xFF890808)],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              stops: [0.0, 0.6],
            ),
          ),
          child: Column(
            children: <Widget>[
              SizedBox(
                height: 250.0,
              ),
              Text(
                valueOnTextWidget.round().toString(),
                style: TextStyle(color: Colors.white, fontSize: 150.0),
              ),
              SizedBox(
                height: 80.0,
              ),
              AwesomeSlider(
                value: valueOnTextWidget,
                min: 0.0,
                max: 100.0,
                thumbColor: Color(0xFF890808),
                roundedRectangleThumbRadius: 80.0,
                thumbSize: 100.0,
                topLeftShadow: true,
                topLeftShadowColor: Colors.redAccent,
                topLeftShadowBlur: MaskFilter.blur(BlurStyle.normal, 8.0),
                bottomRightShadow: true,
                bottomRightShadowColor: Colors.black,
                bottomRightShadowBlur: MaskFilter.blur(BlurStyle.normal, 7.0),
                activeLineStroke: 2.0,
                activeLineColor: Colors.redAccent,
                inactiveLineColor: Colors.white,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    Icon(
                      Icons.arrow_back,
                      color: Colors.white,
                      size: 28.0,
                    ),
                    SizedBox(width: 10.0),
                    Icon(
                      Icons.arrow_forward,
                      color: Colors.white,
                      size: 28.0,
                    )
                  ],
                ),
                onChanged: (double value) {
                  setState(() {
                    valueOnTextWidget = value;
                  });
                },
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```
</details>



### Awesome Slider Parameters

| Parameter                 |                       Default                       | Description                                                                                                             |
| :------------------------ | :-------------------------------------------------: | :---------------------------------------------------------------------------------------------------------------------- |
| **value** *double*                   |                   @required                         | Value of the Slider Position                                              |
|                                      |                                                     | (**value**!=null)                                                         |
|                                      |                                                     |(**value** >= **min** && **value** <= **max**)                             |
|                                      |                                                     |                                                                           |
| **min** *double*                     |                   @required                         | minimum value for the Slider                                              |
|                                      |                                                     | (**min** !=null)                                                          |
|                                      |                                                     |(**min** <= **max**)                                                       |
|                                      |                                                     |                                                                           |
| **max** *double*                     |                   @required                         | maximum value for the Slider                                              |
|                                      |                                                     | (**max** != null)                                                         |
|                                      |                                                     |                                                                           |
| **onChanged** *ValueChanged **double*** |                                                  |    Called when the user starts selecting a new **value** for the slider.  |
|                                      |                                                     | Parameter for the **onChanged** must be a double                          |
|                                      |                                                     |                                                                           |
| **child** *widget*                   |                                                     | Provide a child Widget to the Slider Thumb.                               |
|                                      |                                                     |                                                                           |
| **sliderWidth** *double*             |Default width will be the Canvas Width with a difference of 40px| The Width of the Slider.                                       |
|                                      |                                                     |                                                                           |
| **thumbSize** *double*               |Default value will be a 90px ratio of the original Canvas it was created| Size of the thumb                                      |
|                                      |                                                     |                                                                           |
| **thumbColor** *Color*               |               Colors.grey                           | Colour of the thumb                                                       |
|                                      |                                                     |                                                                           |
| **roundedRectangleThumbRadius** *double*|                   0.0                            | Radius of Rounded Rectangle                                               |
|                                      |                                                     | more the radius, The square thumb turns to a Circle                        |
|                                      |                                                     |                                                                           |
| **inactiveLineColor** *Color*        |                Colors.blue                          | The color for the inactive portion of the slider track.                   |
|                                      |                                                     |                                                                           |
| **inactiveLineStroke** *double*      |                      4.0                            | The stroke value for the inactive portion of the slider track.            |
|                                      |                                                     | Value for **inactiveLineStroke** = **activeLineStroke** unless given different values for both|
|                                      |                                                     |                                                                           |
| **activeLineColor** *Color*          |                Colors.blue                          | The color for the active portion of the slider track.                     |
|                                      |                                                     |                                                                           |
| **activeLineStroke** *double*        |                      4.0                            | The stroke value for the active portion of the slider track.            |
|                                      |                                                     | Value for **activeLineStroke** = **inactiveLineStroke** unless given different values for both|
|                                      |                                                     |                                                                           |
| **topLeftShadow** *bool*             |                    false                            | Give true value if a Shadow required on Top - Left of the thumb           |
|                                      |                                                     |                                                                           |
| **topLeftShadowColor** *Color*       |                    Colors.blueGrey                  | Colour of shadow of Top - Left of the thumb                               |
|                                      |                                                     |                                                                           |
| **topLeftShadowBlur** *MaskFilter*   |       MaskFilter.blur(BlurStyle.normal, 3.0)        | MaskFilter blur value for shadow of Top - Left of the thumb               |
|                                      |                                                     |                                                                           |
| **bottomRightShadow** *bool*         |                    false                            | Give true value if a Shadow required on Bottom - Right of the thumb       |
|                                      |                                                     |                                                                           |
| **bottomRightShadowColor** *Color*   |                    Colors.blueGrey                  | Colour of shadow of Bottom - Right of the thumb                           |
|                                      |                                                     |                                                                           |
| **bottomRightShadowBlur** *MaskFilter*|       MaskFilter.blur(BlurStyle.normal, 3.0)       | MaskFilter blur value for shadow of Bottom - Right of the thumb           |








CHANGELOG.md
## [0.0.1] - 02 April 2020

* Added Functions such as topLeftShadow, bottomRightShadow and a child property.


example/awesome_slider_example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:awesomesliderexample/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MainApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/src/awesome_slider_main.dart
import 'package:flutter/cupertino.dart';
import 'dart:ui';
import 'package:flutter/material.dart';
import 'awesome_slider_painter.dart';

class AwesomeSlider extends StatefulWidget {
  AwesomeSlider({
    this.value,
    this.min,
    this.max,
    this.onChanged,
    this.child,
    this.sliderWidth,
    this.thumbSize,
    this.thumbColor = Colors.grey,
    this.roundedRectangleThumbRadius = 0.0,
    this.inactiveLineColor = Colors.blue,
    this.inactiveLineStroke,
    this.activeLineColor = Colors.blue,
    this.activeLineStroke,
    this.topLeftShadow = false,
    this.topLeftShadowColor = Colors.blueGrey,
    this.topLeftShadowBlur,
    this.bottomRightShadow = false,
    this.bottomRightShadowColor = Colors.blueGrey,
    this.bottomRightShadowBlur,
  })  : assert(value != null),
        assert(min != null),
        assert(max != null),
        assert(min <= max),
        assert(value >= min && value <= max);

  /// Value of the Slider Position
  /// (Value!=null)
  /// (value >= min && value <= max)

  final double value;

  /// minimum value for the Slider
  /// (min != null)
  /// (min <= max)
  final double min;

  /// maximum value for the Slider
  /// (max != null)
  final double max;

  /// Called when the user starts selecting a new value for the slider.
  final ValueChanged<double> onChanged;

  /// Provide a child Widget to the Slider Thumb
  final Widget child;

  /// Total Width of the Slider.
  /// Default width will be the Canvas Width with a difference of 40px
  final double sliderWidth;

  /// Size of the thumb
  /// Default value will be a 90px ratio of the original Canvas
  final double thumbSize;

  ///Colour of the thumb
  ///Default colour is grey Colour
  final Color thumbColor;

  /// Give this radius to convert the Rectangle into a Rounded Rectangle
  /// Increase in radius will make the  rectangle into a circle
  final double roundedRectangleThumbRadius;

  ///The color for the inactive portion of the slider track.
  ///Default colour is Blue Colour
  final Color inactiveLineColor;

  ///The stroke value for the inactive portion of the slider track.
  ///Default stroke value is 4.0
  ///Value for inactiveLineStroke = activeLineStroke unless given different values for both
  final double inactiveLineStroke;

  ///The color for the active portion of the slider track.
  ///Default colour is Blue Colour
  final Color activeLineColor;

  ///The stroke value for the active portion of the slider track.
  ///Default stroke value is 4.0
  ///Value for activeLineStroke = inactiveLineStroke unless given different values for both
  final double activeLineStroke;

  ///Give true value if a Shadow required on Top - Left of the thumb
  final bool topLeftShadow;

  ///Colour of shadow of Top - Left of the thumb
  ///Default is Blue - Grey
  final Color topLeftShadowColor;

  ///MaskFilter blur value for shadow of Top - Left of the thumb
  ///MaskFilter.blur(BlurStyle.normal, 3.0)
  final MaskFilter topLeftShadowBlur;

  ///Give true value if a Shadow required on Bottom - Right of the thumb
  final bool bottomRightShadow;

  ///Colour of shadow of Bottom - Right of the thumb
  ///Default is Blue - Grey
  final Color bottomRightShadowColor;

  ///MaskFilter blur value for shadow of Top - Left of the thumb
  ///MaskFilter.blur(BlurStyle.normal, 3.0)
  final MaskFilter bottomRightShadowBlur;

  @override
  _AwesomeSliderState createState() => _AwesomeSliderState();
}

class _AwesomeSliderState extends State<AwesomeSlider> {
  double sliderXCoordinatePositionNow = 0.0;

  double _strokeOfInactiveLine() => (widget.inactiveLineStroke == null)
      ? (widget.activeLineStroke == null) ? 4.0 : widget.activeLineStroke
      : widget.inactiveLineStroke;

  double _strokeOfActiveLine() => (widget.activeLineStroke == null)
      ? (widget.inactiveLineStroke == null) ? 4.0 : widget.inactiveLineStroke
      : widget.activeLineStroke;

  MaskFilter _topLeftShadowBlur() => (widget.topLeftShadowBlur == null)
      ? MaskFilter.blur(BlurStyle.normal, 3.0)
      : widget.topLeftShadowBlur;

  MaskFilter _bottomRightShadowBlur() => (widget.bottomRightShadowBlur == null)
      ? MaskFilter.blur(BlurStyle.normal, 3.0)
      : widget.bottomRightShadowBlur;

  double _incrementValueForThumb() =>
      (widget.value == 0.0) ? widget.min : widget.value - widget.min;

  double _lineLengthForPixel() => _sliderWidth() - _sliderHeight();
  double _userValueForPixel() => widget.max - widget.min;
  double _pixelDivision() => _lineLengthForPixel() / _userValueForPixel();

  double _sliderChildPosition() => _incrementValueForThumb() * _pixelDivision();

  double _sliderWidth() {
    double userInputWidth = widget.sliderWidth;
    double screenWidth = window.physicalSize.width;
    double pixelRatio = window.devicePixelRatio;
    double sliderWidth = (userInputWidth == null)
        ? ((screenWidth / pixelRatio) - 40.0)
        : userInputWidth;
    return sliderWidth;
  }

  double _sliderHeight() {
    double userInputHeight = widget.thumbSize;
    double screenHeight = window.physicalSize.height;
    double pixelRatio = window.devicePixelRatio;
    double multiplicationFactor = (90 / 805.3333334);
    double sliderHeight = (userInputHeight == null)
        ? ((screenHeight / pixelRatio) * multiplicationFactor)
        : userInputHeight;
    return sliderHeight.roundToDouble();
  }

  void _onDragUpdate(DragUpdateDetails dragUpdateDetails) {
    Offset localDragUpdate = dragUpdateDetails.localPosition;
    double xCoordinate;
    (localDragUpdate.dx < 0)
        ? xCoordinate = 0
        : (localDragUpdate.dx > _sliderWidth())
            ? xCoordinate = _sliderWidth()
            : xCoordinate = localDragUpdate.dx;
    setState(() {
      sliderXCoordinatePositionNow = xCoordinate;
    });
  }

  void _onDragStart(DragStartDetails dragStartDetails) {
    Offset localDragStart = dragStartDetails.localPosition;
    double xCoordinate;
    (localDragStart.dx < 0)
        ? xCoordinate = 0
        : (localDragStart.dx > _sliderWidth())
            ? xCoordinate = _sliderWidth()
            : xCoordinate = localDragStart.dx;
    setState(() {
      sliderXCoordinatePositionNow = xCoordinate;
    });
  }

  void _value() {
    double incrementValue = sliderXCoordinatePositionNow / _pixelDivision();
    double value = (incrementValue > _userValueForPixel())
        ? _userValueForPixel()
        : incrementValue;
    double userValue = value + widget.min;
    if (widget.onChanged != null) {
      widget.onChanged(userValue);
    }
  }

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      behavior: HitTestBehavior.translucent,
      onHorizontalDragUpdate: (DragUpdateDetails updateDetails) {
        _onDragUpdate(updateDetails);
        _value();
      },
      onHorizontalDragStart: (DragStartDetails startDetails) {
        _onDragStart(startDetails);
        _value();
      },
      child: Container(
        padding: EdgeInsets.all(20.0),
        child: Stack(
          children: <Widget>[
            Container(
              height: _sliderHeight(),
              width: _sliderWidth(),
              child: CustomPaint(
                painter: AwesomeSliderPaint(
                  sliderLength: _sliderWidth(),
                  thumbSize: _sliderHeight(),
                  thumbColor: widget.thumbColor,
                  value: _incrementValueForThumb(),
                  min: widget.min,
                  max: widget.max,
                  inactiveLineColor: widget.inactiveLineColor,
                  inactiveLineStroke: _strokeOfInactiveLine(),
                  activeLineColor: widget.activeLineColor,
                  activeLineStroke: _strokeOfActiveLine(),
                  currentTouchPosition: sliderXCoordinatePositionNow,
                  roundedThumbRadius: widget.roundedRectangleThumbRadius,
                  topLeftShadowColor: widget.topLeftShadowColor,
                  bottomRightShadowColor: widget.bottomRightShadowColor,
                  topLeftShadowBlurFactor: _topLeftShadowBlur(),
                  bottomRightShadowBlurFactor: _bottomRightShadowBlur(),
                  bottomRightShadow: widget.bottomRightShadow,
                  topLeftShadow: widget.topLeftShadow,
                ),
              ),
            ),
            Positioned(
              height: _sliderHeight(),
              width: _sliderHeight(),
              left: _sliderChildPosition(),
              child: Container(
                height: double.infinity,
                width: double.infinity,
                child: widget.child,
              ),
            ),
          ],
        ),
      ),
    );
  }
}


lib/src/awesome_slider_painter.dart
import 'dart:ui';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

class AwesomeSliderPaint extends CustomPainter {
  AwesomeSliderPaint({
    @required this.sliderLength,
    @required this.thumbSize,
    @required this.thumbColor,
    @required this.value,
    @required this.min,
    @required this.max,
    @required this.roundedThumbRadius,
    @required this.inactiveLineColor,
    @required this.inactiveLineStroke,
    @required this.currentTouchPosition,
    @required this.activeLineColor,
    @required this.activeLineStroke,
    @required this.topLeftShadowColor,
    @required this.bottomRightShadowColor,
    @required this.topLeftShadowBlurFactor,
    @required this.bottomRightShadowBlurFactor,
    @required this.topLeftShadow,
    @required this.bottomRightShadow,
  });
  final double sliderLength;
  final double thumbSize;
  final Color thumbColor;
  final double value;
  final double min;
  final double max;
  final double roundedThumbRadius;
  final Color inactiveLineColor;
  final double inactiveLineStroke;
  final double currentTouchPosition;
  final Color activeLineColor;
  final double activeLineStroke;
  final Color topLeftShadowColor;
  final Color bottomRightShadowColor;
  final MaskFilter bottomRightShadowBlurFactor;
  final MaskFilter topLeftShadowBlurFactor;
  final bool topLeftShadow;
  final bool bottomRightShadow;

  @override
  void paint(Canvas canvas, Size size) {
    double roundedRectangleTopLeftShadowShift = 13.0;
    double roundedRectangleBottomRightShadowShift = 15.0;
    double _increment() {
      double lineLengthForPixel = sliderLength - thumbSize;
      double userValueForPixel = max - min;
      double pixelDivision = lineLengthForPixel / userValueForPixel;
      return value * pixelDivision;
    }

    ///    Inactive Line Paint

    Path inactiveLinePath = Path();
    Paint inactiveLinePaint = Paint()
      ..color = inactiveLineColor
      ..style = PaintingStyle.stroke
      ..strokeWidth = inactiveLineStroke;
    inactiveLinePath.moveTo(0.0, thumbSize / 2);
    inactiveLinePath.lineTo(sliderLength, thumbSize / 2);
    canvas.drawPath(inactiveLinePath, inactiveLinePaint);

    ///    Active Line Paint

    Path activeLinePath = Path();
    Paint activeLinePaint = Paint()
      ..color = activeLineColor
      ..style = PaintingStyle.stroke
      ..strokeWidth = activeLineStroke;
    activeLinePath.moveTo(0.0, thumbSize / 2);
    activeLinePath.lineTo(currentTouchPosition, thumbSize / 2);
    canvas.drawPath(activeLinePath, activeLinePaint);

    ///    Rounded Rectangle Top Left Shadow Paint

    Path roundedRectangleTopLeftShadow = Path();
    Paint roundedRectangleTopLeftShadowPaint = Paint()
      ..color = topLeftShadowColor
      ..maskFilter = topLeftShadowBlurFactor;

    roundedRectangleTopLeftShadow.addRRect(RRect.fromLTRBR(
        0.0 - roundedRectangleTopLeftShadowShift + _increment(),
        0.0 - roundedRectangleTopLeftShadowShift,
        thumbSize - roundedRectangleTopLeftShadowShift + _increment(),
        thumbSize - roundedRectangleTopLeftShadowShift,
        Radius.circular(roundedThumbRadius)));
    if (topLeftShadow == true)
      canvas.drawPath(
          roundedRectangleTopLeftShadow, roundedRectangleTopLeftShadowPaint);

    ///    Rounded Rectangle Bottom Right Shadow Paint

    Path roundedRectangleBottomRightShadow = Path();
    Paint roundedRectangleBottomRightShadowPaint = Paint()
      ..color = bottomRightShadowColor
      ..maskFilter = bottomRightShadowBlurFactor;
    roundedRectangleBottomRightShadow.addRRect(RRect.fromLTRBR(
        0.0 + roundedRectangleBottomRightShadowShift + _increment(),
        0.0 + roundedRectangleBottomRightShadowShift,
        thumbSize + roundedRectangleBottomRightShadowShift + _increment(),
        thumbSize + roundedRectangleBottomRightShadowShift,
        Radius.circular(roundedThumbRadius)));
    if (bottomRightShadow == true)
      canvas.drawPath(roundedRectangleBottomRightShadow,
          roundedRectangleBottomRightShadowPaint);

    ///   Rounded Rectangle Thumb Paint

    Path roundedRectangle = Path();
    Paint roundedRectanglePaint = Paint()..color = thumbColor;

    roundedRectangle.addRRect(RRect.fromLTRBR(
        0.0 + _increment(),
        0.0,
        thumbSize + _increment(),
        thumbSize,
        Radius.circular(roundedThumbRadius)));
    canvas.drawPath(roundedRectangle, roundedRectanglePaint);
  }

  @override
  bool shouldRepaint(CustomPainter oldDelegate) {
    return true;
  }
}


example/awesome_slider_example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:awesome_slider/awesome_slider.dart';

void main() => runApp(MainApp());

class MainApp extends StatefulWidget {
  @override
  _MainAppState createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  double valueOnTextWidget = 0.0;
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Container(
          decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: [
                Colors.lightBlueAccent,
                Color(0xFF100887).withOpacity(0.95)
              ],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              stops: [0.0, 0.5],
            ),
          ),
          child: Column(
            children: <Widget>[
              SizedBox(
                height: 250.0,
              ),
              Text(
                valueOnTextWidget.round().toString(),
                style: TextStyle(color: Colors.white, fontSize: 150.0),
              ),
              SizedBox(
                height: 80.0,
              ),
              AwesomeSlider(
                value: valueOnTextWidget,
                min: 0.0,
                max: 100.0,
                thumbColor: Color(0xFF100887),
                roundedRectangleThumbRadius: 25.0,
                thumbSize: 100.0,
                topLeftShadow: true,
                topLeftShadowColor: Colors.lightBlueAccent,
                topLeftShadowBlur: MaskFilter.blur(BlurStyle.normal, 11.0),
                bottomRightShadow: true,
                bottomRightShadowColor: Colors.white.withOpacity(0.5),
                bottomRightShadowBlur: MaskFilter.blur(BlurStyle.normal, 11.0),
                activeLineStroke: 2.0,
                activeLineColor: Colors.blueAccent,
                inactiveLineColor: Colors.white,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    Icon(
                      Icons.arrow_back_ios,
                      color: Colors.white,
                      size: 28.0,
                    ),
                    SizedBox(width: 10.0),
                    Icon(
                      Icons.arrow_forward_ios,
                      color: Colors.white,
                      size: 28.0,
                    )
                  ],
                ),
                onChanged: (double value) {
                  setState(() {
                    valueOnTextWidget = value;
                  });
                },
              ),
            ],
          ),
        ),
      ),
    );
  }
}
